compiler/rustc_parse/src/parser/path.rs

```diff
@@ -449,7 +449,11 @@ impl<'a> Parser<'a> {
         prev_token_before_parsing: Token,
         error: &mut Diag<'_>,
     ) {
-        if ((style == PathStyle::Expr && self.parse_paren_comma_seq(|p| p.parse_expr()).is_ok())
+        if ((style == PathStyle::Expr
```

Comment:
I've pondered this in the original PR, too. We might be able to make this check more readable by rewriting it into sth. like that:

```rs
if !match style {
    PathStyle::Expr => /* parse seq is ok */,
    PathStyle::Pat => /* parse seq is ok */,
    _ => false,
} {
    return;
}
let (token::ModSep | token::RArrow) = self.token.kind else { return };
```

However that might be even more confusing lol ^^

Reply:
Or

```rs
match style {
    PathStyle::Expr if let Ok(_) = /* parse seq */ => {}
    PathStyle::Pat if let Ok(_) = /* parse seq */ => {}
    _ => return,
}
let (token::ModSep | token::RArrow) = self.token.kind else { return };
/* span suggestion verbose */
```

Reply:
No need to apply my suggestions if they turn out to be illegible! Just throwing something out there :)

Reply:
Yes, I agree, the `if` has grown into a chungus. I kinda like the `match-if-let` variant, even though the empty body of the match arm kinda bothers me. But it looks more readable to me.

Also, shouldn't the token deconstruction should be more like this?
```rust
let (token::ModSep | token::RArrow) = self.token.kind else { 
    /* span suggestion verbose */
   return;
 };
```
Because we do not want to emit the suggestion when `ModSep` or `RArrow` is present.